bin/compile

```
@@ -45,15 +45,95 @@ ENVSCRIPT=$BUILD_DIR/.profile.d/tesseract-ocr.sh
 
 echo "-----> Building runtime environment for Tesseract-OCR"
 mkdir -p $BUILD_DIR/.profile.d
-echo "export PATH=\"$HOME/$TESSERACT_VENDOR_PATH/bin:\$PATH\"" > $ENVSCRIPT
-echo "export LD_LIBRARY_PATH=\"$HOME/$TESSERACT_VENDOR_PATH/lib:\$LD_LIBRARY_PATH\"" >> $ENVSCRIPT
-echo "export TESSDATA_PREFIX=\"$HOME/$TESSERACT_DATA_PATH\"" >> $ENVSCRIPT
+echo "export PATH=\"/app/$TESSERACT_VENDOR_PATH/bin:\$PATH\"" > $ENVSCRIPT
+echo "export LD_LIBRARY_PATH=\"/app/$TESSERACT_VENDOR_PATH/lib:\$LD_LIBRARY_PATH\"" >> $ENVSCRIPT
+echo "export TESSDATA_PREFIX=\"/app/$TESSERACT_DATA_PATH\"" >> $ENVSCRIPT
+
+# S3 Download Functions
+
+# Download from S3 using native curl --aws-sigv4
+download_s3() {
+  local bucket="$1"
+  local key="$2"
+  local output_file="$3"
+  local region="${AWS_REGION:-us-east-1}"
+
+  curl -f -s --user "${ML_MODELS_BUCKET_KEY_ID}:${ML_MODELS_BUCKET_KEY_SECRET}" \
+       --aws-sigv4 "aws:amz:${region}:s3" \
+       -o "$output_file" \
+       "https://${bucket}.s3.${region}.amazonaws.com/${key}"
+}
+
+# List S3 objects using native curl --aws-sigv4
+list_s3_objects() {
+  local bucket="$1"
+  local prefix="${2:-}"
+  local region="${AWS_REGION:-us-east-1}"
+
+  curl -s --user "${ML_MODELS_BUCKET_KEY_ID}:${ML_MODELS_BUCKET_KEY_SECRET}" \
```

We should probably add the `-f` flag to fail fast in the case that an error is encountered during the transfer.

May also want to add the `-S` flag so an error message gets printed to the logs if/when curl fails so its easier to track down the issue.
```suggestion
  curl -f -s -S --user "${ML_MODELS_BUCKET_KEY_ID}:${ML_MODELS_BUCKET_KEY_SECRET}" \
```